### PR TITLE
GenAPI: update event declaration and reference parameters.

### PR DESCRIPTION
**Pull request by andriipatsula**
Epic: https://github.com/dotnet/arcade/issues/10291

### CODE REVIEW

File: src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxRewriter/BodyBlockCSharpSyntaxRewriter.cs [hunk 1]
```diff
@@ -48,7 +48,9 @@ public BodyBlockCSharpSyntaxRewriter(string? exceptionMessage)
             }
 
             string returnType = rs.ReturnType.ToString();
-            if (returnType != "void" && returnType != "System.Void")
+            if (returnType != "void" && returnType != "System.Void" ||
```
Comment (ViktorHofer) [inline src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxRewriter/BodyBlockCSharpSyntaxRewriter.cs:51]: This is easy to get/read wrong. Please add braces. I assume it should be `((returnType != "void" && returnType != "System.Void") || X)`

**Review by ViktorHofer — APPROVED** [review]

File: src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs [hunk 2]
```diff
@@ -66,6 +66,19 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 }
             }
 
+            if (symbol.Kind == SymbolKind.Event)
```
Comment (smasher164) [inline src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs:69]: You can probably replace this `if` with `if (symbol is IEventSymbol eventSymbol)`, and avoid the nested conditional

File: src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxRewriter/BodyBlockCSharpSyntaxRewriter.cs (hunk 1 above)
Comment (andriipatsula) [reply src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxRewriter/BodyBlockCSharpSyntaxRewriter.cs:51]: Agree. Thanks

File: src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs (hunk 2 above)
Comment (andriipatsula) [reply src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs]: Updated:
```C#
if (symbol is IEventSymbol eventSymbol && !eventSymbol.IsAbstract) {...}
```

**Pull request merged**